code-samples/creating-your-bot/event-handling/events/guildMemberUpdate.js

```diff
@@ -1,21 +1,48 @@
 module.exports = {
 	name: "guildMemberUpdate",
 	async execute(oldMember, newMember) {
-		console.log(oldMember, newMember);
-		// if (!interaction.isChatInputCommand()) return;
+		//console.log(oldMember, newMember);
+    
+    //var defaultHost = "https://forum.citydao.io/";
+    
+    /* 
+    / Grab object for the new role
+    / newRoleId = ID of the changed Role
+    / newRoleName = Name of the changed Role
+    */
+    var newRole = newMember.roles.cache.difference(oldMember.roles.cache).last();
```

Comment:
Some info on the cache https://github.com/discordjs/discord.js/discussions/7661#discussioncomment-2356026
This way is perfectly reasonable but something to keep in mind